### DISCUSSION TITLE
Git cannot clone from China

### DISCUSSION THREAD

**Discussion by howarduong** [vladmandic/sdnext#790] [Q&A]
### Issue Description

hi buddy, my first installation test, found these problems, unable to install successfully, I wonder if there is a solution.
Please reply in your spare time! I would appreciate it very much

22:47:37-586308 INFO     Installing package:
                         git+https://github.com/openai/CLIP.git@d50d76daa670286dd6cacf3bcd80b5e4823fc8e1
22:47:59-723392 ERROR    Error running pip: install --upgrade
                         git+https://github.com/openai/CLIP.git@d50d76daa670286dd6cacf3bcd80b5e4823fc8e1
22:47:59-727448 INFO     Installing repositories
22:49:13-432503 ERROR    Error running git: None / clone "https://github.com/crowsonkb/k-diffusion.git" "D:\Vlad
                         Diffusion AI\automatic\repositories\k-diffusion"
22:49:13-482934 ERROR    Error running git: None / -C "D:\Vlad Diffusion AI\automatic\repositories\k-diffusion"
                         checkout b43db16749d51055f813255eea2fdf1def801919
22:49:34-683032 ERROR    Error running git: None / clone "https://github.com/sczhou/CodeFormer.git" "D:\Vlad Diffusion
                         AI\automatic\repositories\CodeFormer"
22:49:34-736781 ERROR    Error running git: None / -C "D:\Vlad Diffusion AI\automatic\repositories\CodeFormer" checkout
                         c5b4593074ba6214284d6acd5f1719b6c5d739af
22:49:55-928763 ERROR    Error running git: None / clone "https://github.com/salesforce/BLIP.git" "D:\Vlad Diffusion
                         AI\automatic\repositories\BLIP"
22:49:55-977738 ERROR    Error running git: None / -C "D:\Vlad Diffusion AI\automatic\repositories\BLIP" checkout
                         48211a1594f1321b00f14c9f7a5b4813144b2fb9
22:49:55-983600 INFO     Installing submodules
22:53:38-897604 ERROR    Error running git: None / submodule --quiet update --init --recursive
22:53:38-904181 INFO     Updating submodules
22:54:01-288751 ERROR    Error running git: extensions-builtin/multidiffusion-upscaler-for-automatic1111 / pull origin
                         main --autostash --rebase --force
22:54:29-585021 ERROR    Error running git: modules/lora / pull origin main --autostash --rebase --force
22:54:29-587557 INFO     Extensions enabled: ['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'prompt-bracket-checker', 'ScuNET',
                         'sd-dynamic-thresholding', 'sd-extension-aesthetic-scorer', 'sd-extension-steps-animation',
                         'sd-extension-system-info', 'sd-webui-controlnet', 'sd-webui-model-converter', 'seed_travel',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR']
22:55:05-605479 ERROR    Error running git: D:\Vlad Diffusion
                         AI\automatic\extensions-builtin\multidiffusion-upscaler-for-automatic1111 / pull origin main
                         --autostash --rebase --force
22:55:26-991037 ERROR    Error running git: D:\Vlad Diffusion AI\automatic\extensions-builtin\sd-dynamic-thresholding /
                         pull origin master --autostash --rebase --force
22:56:08-656419 ERROR    Error running git: D:\Vlad Diffusion
                         AI\automatic\extensions-builtin\stable-diffusion-webui-images-browser / pull origin main
                         --autostash --rebase --force
22:56:57-482477 ERROR    Error running git: D:\Vlad Diffusion
                         AI\automatic\extensions-builtin\stable-diffusion-webui-rembg / pull origin master --autostash
                         --rebase --force
22:57:14-573407 INFO     Extensions enabled: []
22:57:14-576411 INFO     Updating Wiki
22:57:14-578415 WARNING  Setup complete with errors (14)
22:57:14-580412 WARNING  See log file for more details: setup.log
22:57:14-586956 INFO     Running extension preloading
22:57:14-624606 INFO     Server arguments: []
Warning: CodeFormer not found at path D:\Vlad Diffusion AI\automatic\repositories\CodeFormer\inference_codeformer.py
Warning: BLIP not found at path D:\Vlad Diffusion AI\automatic\repositories\BLIP\models\blip.py
Warning: k_diffusion not found at path D:\Vlad Diffusion AI\automatic\repositories\k-diffusion\k_diffusion\sampling.py
No module 'xformers'. Proceeding without it.
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ D:\Vlad Diffusion AI\automatic\launch.py:141 in <module>                                                            │
│                                                                                                                     │
│   140                                                                                                               │
│ > 141     instance = start_server(immediate=True, server=None)                                                      │
│   142     while True:                                                                                               │
│                                                                                                                     │
│ D:\Vlad Diffusion AI\automatic\launch.py:116 in start_server                                                        │
│                                                                                                                     │
│   115     server = importlib.util.module_from_spec(module_spec)                                                     │
│ > 116     module_spec.loader.exec_module(server)                                                                    │
│   117     if args.test:                                                                                             │
│ in exec_module:883                                                                                                  │
│ in _call_with_frames_removed:241                                                                                    │
│                                                                                                                     │
│ D:\Vlad Diffusion AI\automatic\webui.py:34 in <module>                                                              │
│                                                                                                                     │
│    33 import ldm.modules.encoders.modules # pylint: disable=W0611,C0411                                             │
│ >  34 from modules import extra_networks, ui_extra_networks_checkpoints # pylint: disable=C041                      │
│    35 from modules import extra_networks_hypernet, ui_extra_networks_hypernets, ui_extra_netwo                      │
│                                                                                                                     │
│ D:\Vlad Diffusion AI\automatic\modules\ui_extra_networks_checkpoints.py:5 in <module>                               │
│                                                                                                                     │
│    4                                                                                                                │
│ >  5 from modules import shared, ui_extra_networks, sd_models                                                       │
│    6                                                                                                                │
│                                                                                                                     │
│ D:\Vlad Diffusion AI\automatic\modules\ui_extra_networks.py:11 in <module>                                          │
│                                                                                                                     │
│    10 from modules import shared                                                                                    │
│ >  11 from modules.images import read_info_from_image                                                               │
│    12 from modules.generation_parameters_copypaste import image_from_url_text                                       │
│                                                                                                                     │
│ D:\Vlad Diffusion AI\automatic\modules\images.py:16 in <module>                                                     │
│                                                                                                                     │
│    15                                                                                                               │
│ >  16 from modules import sd_samplers, shared, script_callbacks, errors                                             │
│    17 from modules.shared import opts, log                                                                          │
│                                                                                                                     │
│ D:\Vlad Diffusion AI\automatic\modules\sd_samplers.py:1 in <module>                                                 │
│                                                                                                                     │
│ >  1 from modules import sd_samplers_compvis, sd_samplers_kdiffusion, shared                                        │
│    2                                                                                                                │
│                                                                                                                     │
│ D:\Vlad Diffusion AI\automatic\modules\sd_samplers_kdiffusion.py:4 in <module>                                      │
│                                                                                                                     │
│     3 import torch                                                                                                  │
│ >   4 import k_diffusion.sampling                                                                                   │
│     5 from modules import prompt_parser, devices, sd_samplers_common                                                │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
ModuleNotFoundError: No module named 'k_diffusion'

### Version Platform Description

windows11 system, inter 9400F cpu, amdRX570GPU, 8G memory, Chinese environment, Mainland China

**Comment by vladmandic** [comment]
basically, not a a sigle git operation completed sucessfully.
what is the output of `git --version`?
can you run any manual `git clone` operation?

**Comment by howarduong** [comment]
I think it will not be the reason for Chinese mainland for network restrictions, if so, I have no solution, maybe set up a VPN, this is my version of git, you see
C:\Users\Uong>git --version
git version 2.40.1.windows.1

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: Vladimir ***@***.***>
发送时间: 2023年5月6日 23:42
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [vladmandic/automatic] [Issue]: (Issue #788)


basically, not a a sigle git operation completed sucessfully.
what is the output of git --version?
can you run any manual git clone operation?

―

**Comment by vladmandic** [comment]
there are some users running from mainland china using github proxy services. i'm totally unfamilar with that, perhaps someone can ask if you ask in discussions?
